futures-util/src/stream/catch_unwind.rs

```diff
@@ -45,7 +45,6 @@ impl<S> Stream for CatchUnwind<S>
             }
             CatchUnwindState::Stream(stream) => stream,
         };
-        // FIXME: fix passing the actual context here.
         let res = catch_unwind(|| (stream.poll(&mut task::Context::panicking()), stream));
```

Comment:
This needs more thought.

Reply:
As mentioned in [this comment], unwind safety seems to have been deferred to implementors of NotifyHandle and the platform-specific task-local storage implementation. This also was the case for TLS-based contexts. So it seem ignoring this using `panic::AssertUnwindSafe` is the way to go, and making sure panic safety is a documented requirement for `NotifyHandle`, `UnsafeNotify`, and task-local.

[this comment]: https://github.com/rust-lang-nursery/futures-rs/pull/744#issuecomment-364449525